### ISSUE TITLE
[BUG] [JAVA] Missing imports in okhttp-gson client

### ISSUE THREAD

**Issue by swanysimon** [OpenAPITools/openapi-generator#14599]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue? https://github.com/swanysimon/java-generator-example-14599
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

I'm trying to upgrade my Gradle plugin version from 5.4.0 to 6.3.0 and running into an issue with generated enum classes for my API client.

There are a number of missing imports in https://github.com/juanavilactn/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/pojo.mustache. This is the list I get in my IDE that I have to manually add imports for with my enums:

```
com.google.gson.TypeAdapter
com.google.gson.stream.JsonReader
com.google.gson.stream.JsonWriter
java.io.IOException
```

##### openapi-generator version

6.3.0

##### Generation Details

```
import org.openapitools.generator.gradle.plugin.tasks.GenerateTask

plugins {
    id "java"
    id "org.openapi.generator" version "6.3.0"
}

group = "example.example"
version = "0.0.1"

repositories { mavenCentral() }

java { toolchain { languageVersion.set(JavaLanguageVersion.of(11)) } }

sourceSets { main { java.srcDirs += "${buildDir}/generated/src/main/java" } }

dependencies {
    implementation "com.google.code.gson:gson:2.8.9"
    implementation "com.squareup.okhttp3:logging-interceptor:4.9.3"
    implementation "com.squareup.okhttp3:okhttp:4.9.3"
    implementation "com.squareup.okio:okio:2.8.0"
    implementation "io.gsonfire:gson-fire:1.8.5"
    implementation "io.swagger:swagger-core:1.6.9"
    implementation "javax.annotation:javax.annotation-api:1.3.2"
    implementation "javax.ws.rs:javax.ws.rs-api:2.1.1"
    implementation "org.openapitools:jackson-databind-nullable:0.2.2"
}

def generateSchemaTask = tasks.register("generateJavaClient", GenerateTask) {
    generatorName = "java"
    group = project.group
    version = project.version
    inputSpec = file("openapi.yaml").toString()
    outputDir = "${buildDir}/generated"
    apiPackage = "example.example.api"
    modelPackage = "example.example.api.model"
    configOptions = [
            dateLibrary            : "java8",
            hideGenerationTimestamp: "true",
    ]
}
```

##### Steps to reproduce

Try to generate with the above with an enum object. I can work on a minimal example if it would actually be helpful, but it seems pretty straightforward.

Minimal example: https://github.com/swanysimon/java-generator-example-14599

##### Related issues/PRs

n/a

##### Suggest a fix

I can make this change myself since it is just adding imports, but this clearly made it through testing and its possible folks want a more thorough fix and/or testing.

**Comment by wing328** [comment]
> I can work on a minimal example

please provide one for easier troubleshooting.

**Comment by swanysimon** [comment]
@wing328 Here's a (not) working example: https://github.com/swanysimon/java-generator-example-14599

Updating the original comment too

**Comment by stowns** [comment]
I seem to have the same missing imports

**Comment by wing328** [comment]
@swanysimon if you change the spec to 3.0.2, it seems to work (there's another issue how the parameter in which int64 should be format instead)

**Comment by wing328** [comment]
@stowns please provide a spec to reproduce the issue.

**Comment by swanysimon** [comment]
Thanks for the update @wing328 . 3.0.2 does indeed seem to work, but then I run into issues with a nullable ref a la https://github.com/OpenAPITools/openapi-generator/issues/5180. Will try to get that working after reading through that thread

**Comment by swanysimon** [comment]
Looks like I'm blocked on a problem extremely similar to https://github.com/OpenAPITools/openapi-generator/issues/14041 or https://github.com/OpenAPITools/openapi-generator/issues/12556 and cannot upgrade

**Comment by stowns** [comment]
ah yes @wing328 using `swagger: "2.0"`

**Comment by ole-v-v** [comment]
My 15 lines example here: https://github.com/ole-v-v/openapi-generator-java-bug-14599-example/blob/main/test.swagger.json

**Comment by stowns** [comment]
Unfortunately, migrating my schema yaml from swagger 2.0 -> openapi 3.0.2 didn't solve the import issue. Still missing imports for `IOException` and `TypeAdapter` in my models using `6.3.0`. Unfortunately, I can't share the schema :/ .

**Comment by ole-v-v** [comment]
*Have you tested with the latest master to confirm the issue still exists?* Yes, I have observed the bug on master (up to commit d7edbad8d649d2ebe83c7933124aef58a1f2b7fb from Wed Feb 15 02:12:39 2023 -0500 (inclusive)).

**Comment by vanwobe** [comment]
above PR should fix this issue of missing imports.

**Comment by wing328** [comment]
I've tested the fix (#15388) with the spec provided by @ole-v-v and confirmed the fix. 

Thanks for the PR, which has just been merged. Please pull the latest master to give it a try.

**Comment by ole-v-v** [comment]
In a few tests using the latest from master (up to and including commit b6b888302e8d4a597dfc95d981ae6f10d37b9af8 from Sun May 14 23:06:20 2023 +0800 inclusive) I was unable to reproduce the bug. Thanks for fixing.
